### PR TITLE
Feature/validation messages UI #172

### PR DESCRIPTION
**Pull request by Christopher-Sandbach**
Validation handling on front end

Links to #172

### CODE REVIEW

File: src/UI/state/upload/uploadSlice.tsx [hunk 1]
```diff
@@ -1,20 +1,68 @@
...
 
 export const initialState: () => UploadState = () => ({
   modelFile: null,
   dataFile: null,
   loading: false,
+  validationErrors: [],
   templateList: [],
 });
 
+export function unpackOverlays(map_layers: any) {
```
Comment (andykitch) [inline src/UI/state/upload/uploadSlice.tsx:30]: What's this doing here?

File: src/UI/state/upload/uploadSlice.test.tsx [hunk 2]
```diff
@@ -15,4 +19,65 @@ describe('uploadSlice', () => {
...
+        errorType: 'Incorrect data type',
+        expectedType: 'number',
+        receivedType: 'string',
+      },
+      {
+        key: 'Latitude',
+        row: 1,
+        errorType: 'Required data',
+        expectedType: 'string',
+      },
```
Comment (andykitch) [inline src/UI/state/upload/uploadSlice.test.tsx:49]: Worth sticking another row in here too, to test that they aren't all just stuck in the one row

File: src/UI/state/upload/uploadSlice.tsx (hunk 1 above)
Comment (Christopher-Sandbach) [reply src/UI/state/upload/uploadSlice.tsx:30]: Oh I know what I've done. I used it for reference and have forgotten to remove it

File: src/UI/components/upload/validation/validationItem.test.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { render } from '../../../test_config/render';
+import { fireEvent, screen } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import ValidationConsole from './validationConsole';
+import { AppState } from '../../../state/store';
+import { initialState } from '../../../state/upload/uploadSlice';
+import { useAppSelector } from '../../../state/hooks';
+
+jest.mock('../../../state/hooks', () => ({
```
Comment (andykitch) [inline src/UI/components/upload/validation/validationItem.test.tsx:9]: You shouldn't have to mock the useAppSelector - just pass in the state in the render method with what you want to come out of it.

File: src/UI/components/upload/validation/validationItem.test.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  state = {
+    upload: initialState(),
+  };
+  it('renders validation console with no validation errors', () => {
+    render(<ValidationConsole />, state);
+    expect(screen.getByText('Validation Console')).toBeInTheDocument();
+    expect(useAppSelector).toBeCalled();
+    expect(screen.getByTestId('CheckIcon')).toBeInTheDocument();
+  });
+  it('renders validation console with validation errors', () => {
```
Comment (andykitch) [inline src/UI/components/upload/validation/validationItem.test.tsx:50]: Are these test names correct?

File: src/UI/components/upload/validation/validationItem.test.tsx (hunk 3 above)
Comment (andykitch) [reply src/UI/components/upload/validation/validationItem.test.tsx:9]: It's not being used in this component anyway... Why is it here?

File: src/UI/components/upload/validation/validationItem.test.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    ]),
+}));
+
+describe('Validation component', () => {
+  let state: Partial<AppState>;
+  state = {
+    upload: initialState(),
+  };
+  it('renders validation console with no validation errors', () => {
+    render(<ValidationConsole />, state);
```
Comment (andykitch) [inline src/UI/components/upload/validation/validationItem.test.tsx:45]: Is this the right component?

File: src/UI/components/upload/validation/validationConsole.test.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import { render } from '../../../test_config/render';
+import { screen } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import ValidationConsole from './validationConsole';
+import { AppState } from '../../../state/store';
+import { initialState } from '../../../state/upload/uploadSlice';
+import { useAppSelector } from '../../../state/hooks';
+
+jest.mock('../../../state/hooks', () => ({
+  useAppSelector: jest
```
Comment (andykitch) [inline src/UI/components/upload/validation/validationConsole.test.tsx:10]: Again, we shouldn't have to mock this - see other comment. At least it's being used in this component...

File: src/UI/components/upload/data/Upform.tsx [hunk 7]
```diff
@@ -47,8 +48,16 @@ function Upform() {
 
   return (
     <form>
+      <TemplateDownload />
```
Comment (andykitch) [inline src/UI/components/upload/data/Upform.tsx:51]: I'm not sure about having this above the form, especially as the accordion is gone... I think people will download the template once, then not care about it in subsequent uploads, so we don't want it taking up too much space or attention.

File: src/API/src/utils.ts [hunk 8]
```diff
@@ -56,21 +56,17 @@ export const transformHeaderRow = (
...
-      const newIssueList = [];
-      issueList.map((issue) => {
-        issue = issue.replace(
-          `${config['VA-column']}`,
-          `${config['Template-column']}`,
-        );
-        newIssueList.push(issue);
-      });
-      return newIssueList;
+    console.log(validationIssuesArray);
```
Comment (andykitch) [inline src/API/src/utils.ts:63]: Console.log...

File: src/UI/components/upload/data/Upform.tsx (hunk 7 above)
Comment (Christopher-Sandbach) [reply src/UI/components/upload/data/Upform.tsx:51]: Ray wanted it to move down the page like a story. So he wanted the accordions removed and the templates at the top. So, one goes to upload - they grab the template they want/need - fill it in - upload (moving down the page)

Comment (andykitch) [reply src/UI/components/upload/data/Upform.tsx:51]: Ah, fine

File: src/UI/components/upload/validation/validationItem.test.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        errorType: 'Incorrect data type',
+        expectedType: 'test2',
+        receivedType: 'test2',
+      },
+    ],
+  },
+];
+
+describe('Validation component', () => {
+  let state: Partial<AppState>;
```
Comment (andykitch) [inline src/UI/components/upload/validation/validationItem.test.tsx:37]: I don't think this state is being used now...

**Review by andykitch — APPROVED** [review]

**Pull request merged**